### PR TITLE
Attributes collection can be keyword list or map

### LINKED ISSUE #10: Respond to keyword lists and maps?
The `attrs` and `schema` functions respond to keyword lists: https://github.com/mrmicahcooper/ecto_factory/blob/master/lib/ecto_factory.ex#L48

What do you think about them also responding to a `Map` also @mrmicahcooper?

### PR DESCRIPTION
**Pull request by plicjo**
Closes https://github.com/mrmicahcooper/ecto_factory/issues/10

### CODE REVIEW

File: lib/ecto_factory.ex [hunk 1]
```diff
@@ -46,7 +46,7 @@ defmodule EctoFactory do
   """
   @spec schema(atom() | Ecto.Schema, keyword()) :: map()
   def attrs(factory_name, attrs \\ []) do
-    build_attrs(factory_name, attrs)
+    build_attrs(factory_name, to_keyword_list(attrs))
     |> elem(1)
     |> Enum.into(%{}, fn {k, v} -> {to_string(k), v} end)
```
Comment (mrmicahcooper) [inline lib/ecto_factory.ex:51]: After looking at this. We're really iterating through `attrs` a few times. I bet we could make it more efficient if `build_attrs` took a keyword or a map, and also converted the key to a string.

Wanna  try that?

File: lib/ecto_factory.ex [hunk 2]
```diff
@@ -145,4 +145,9 @@ defmodule EctoFactory do
       {schema} -> {schema, []}
     end
   end
+
+  defp to_keyword_list(map_or_keyword_list) when is_map(map_or_keyword_list) do
+    Enum.map(map_or_keyword_list, fn({key, value}) -> {key, value} end)
+  end
+  defp to_keyword_list(map_or_keyword_list), do: map_or_keyword_list
```
Comment (mrmicahcooper) [inline lib/ecto_factory.ex]: We could eliminate this function by using `Enum.into(map_or_keyword_list, [])`

Comment (plicjo) [reply lib/ecto_factory.ex]: 👍

File: lib/ecto_factory.ex (hunk 1 above)
Comment (plicjo) [reply lib/ecto_factory.ex:51]: I'm fairly certain that I don't understand what you mean.

...but I did try this to convert to string up top:

```elixir
def attrs(factory_name, attrs \\ []) do
  build_attrs(factory_name, Enum.into(attrs, [], fn {k, v} -> {to_string(k), v} end))
  |> elem(1)
  |> Enum.into(%{})
end
```

And I got the following:

```
 1) test build attrs by using defined factory and passed in attributes (EctoFactoryTest)
     test/ecto_factory_test.exs:43
     ** (ArgumentError) expected a keyword list, but an entry in the list is not a two-element tuple with an atom as its first element, got: {"age", 99}
     code: user = EctoFactory.attrs(:user_with_default_username, age: 99)
     stacktrace:
       (elixir 1.11.2) lib/keyword.ex:475: Keyword.keys/1
       (ecto_factory 0.2.0) lib/ecto_factory.ex:119: EctoFactory.build_attrs/2
       (ecto_factory 0.2.0) lib/ecto_factory.ex:49: EctoFactory.attrs/2
       test/ecto_factory_test.exs:44: (test)
```

Some thoughts:
1) I have no idea if I understand your request.
2) I have no idea if Ecto wants atoms (I think it does, dunno how to confirm).

**Comment by plicjo** [comment]
@mrmicahcooper How about now?

**Pull request closed without merging**